### PR TITLE
Feature Toggles: Remove use of boolPtr in FeatureFlag struct

### PR DESCRIPTION
**Pull request by mmandrus**
**What is this feature?**

Changes `AllowSelfServe` in the FeatureToggle struct from a bool pointer to a bool, and updates docs to recommend developers set the property

**Why**

In hindsight, safeguarding against the field being improperly set was unnecessary and not worth the code smell of using bool pointers. 
- If a user sets it on a toggle below GA, it won't matter anyway due to other safeguards in place
- If a user forgets to set it on a GA toggle, it will still be visible in the UI, allowing us to receive feedback to make it editable

### CODE REVIEW

File: pkg/services/featuremgmt/models.go [hunk 1]
```diff
@@ -107,30 +107,32 @@ func (s *FeatureFlagStage) UnmarshalJSON(b []byte) error {
...
+	RequiresLicense bool `json:"requiresLicense,omitempty"` // Must be enabled in the license
+	FrontendOnly    bool `json:"frontend,omitempty"`        // change is only seen in the frontend
+	HideFromDocs    bool `json:"hideFromDocs,omitempty"`    // don't add the values to docs
 
 	// This field is only for the feature management API. To enable your feature toggle by default, use `Expression`.
 	Enabled bool `json:"enabled,omitempty"`
+
+	// These are currently unused
+	DocsURL         string `json:"docsURL,omitempty"`
+	RequiresRestart bool   `json:"requiresRestart,omitempty"` // The server must be initialized with the value
```
Comment (ryantxu) [inline pkg/services/featuremgmt/models.go:135]: ^^ I think we should not `AllowSelfServe` when requires restart is true.  The docs one, we can remove if it is never used

File: pkg/services/featuremgmt/toggles_gen_test.go [hunk 2]
```diff
@@ -42,10 +42,7 @@ func TestFeatureToggleFiles(t *testing.T) {
 			if flag.Name != strings.TrimSpace(flag.Name) {
 				t.Errorf("flag Name should not start/end with spaces.  See: %s", flag.Name)
 			}
-			if flag.Stage == FeatureStageGeneralAvailability && flag.AllowSelfServe == nil {
-				t.Errorf("feature stage FeatureStageGeneralAvailability should have the AllowSelfServe field defined")
-			}
-			if flag.AllowSelfServe != nil && flag.Stage != FeatureStageGeneralAvailability {
+			if flag.AllowSelfServe == true && flag.Stage != FeatureStageGeneralAvailability {
```
Comment (ryantxu) [inline pkg/services/featuremgmt/toggles_gen_test.go:45]:
```suggestion
			if flag.AllowSelfServe && flag.Stage != FeatureStageGeneralAvailability && !flag.RequiresRestart {
```

File: pkg/services/featuremgmt/models.go (hunk 1 above)
Comment (mmandrus) [reply pkg/services/featuremgmt/models.go:135]: We aren't able to make that change right now as _every_ backend toggle requires a restart currently, so we are just ignoring this field and doing the restart in HG via the webhook we discussed. In fact, I'm considering commenting the `RequiresRestart` property out until we actually have support for it. 

We are tracking that here https://github.com/grafana/grafana/issues/78971

File: pkg/services/featuremgmt/toggles_gen_test.go (hunk 2 above)
Comment (mmandrus) [reply pkg/services/featuremgmt/toggles_gen_test.go]: See https://github.com/grafana/grafana/pull/79550#discussion_r1428096011

**Review by ryantxu — APPROVED** [review]

**Review by jcalisto — APPROVED** [review]

**Pull request merged**